bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -164,4 +165,91 @@ private long getArtifactsCountHelper(int artifactTypeID, String whereClause) thr
 			caseDb.releaseSingleUserCaseReadLock();
 		}
 	}
+
+    /**
+     * Determine if an artifact of a given type exists for a given file with a
+     * specific set of attributes.
+     *
+     * @param file          The file whose artifacts need to be looked at.
+     * @param artifactType  The type of artifact to look for.
+     * @param attributesMap The collection of attributes to look for.
+     *
+     * @return True if the specific artifact exists; otherwise false.
+     *
+     * @throws TskCoreException If there is a problem getting artifacts or
+     *                             attributes.
+     */
+    public static boolean checkIfArtifactExists(AbstractFile file, BlackboardArtifact.ARTIFACT_TYPE artifactType,
+            Map<BlackboardAttribute.Type, String> attributesMap) throws TskCoreException {
+
+        ArrayList<BlackboardArtifact> artifactsList;
+
+        /*
+         * Get the file's artifacts.
+         */
+		artifactsList = file.getArtifacts(artifactType);
+		if (artifactsList.isEmpty()) {
+			return false;
+		}
+
+        /*
+         * Get each artifact's attributes and analyze them for matches.
+         */
+        for (BlackboardArtifact artifact : artifactsList) {
+			if (checkIfAttributesMatch(artifact.getAttributes(), attributesMap)) {
+				/*
+				 * The exact artifact exists, so we don't need to look any
+				 * further.
+				 */
+				return true;
+			}
+        }
+
+        /*
+         * None of the artifacts have the exact set of attribute type/value
+         * combinations. The provided file does not have the artifact being
+         * sought.
+         */
+        return false;
+    }
+
+    /**
+     * Determine if the supplied attribute type/value combinations can all be
+     * found in the supplied attributes list.
+     *
+     * @param attributesList The list of attributes to analyze.
+     * @param attributesMap  The attribute type/value combinations to check for.
+     *
+     * @return True if all attributes are found; otherwise false.
+     */
+    private static boolean checkIfAttributesMatch(List<BlackboardAttribute> attributesList, Map<BlackboardAttribute.Type, String> attributesMap) {
+        for (Map.Entry<BlackboardAttribute.Type, String> mapEntry : attributesMap.entrySet()) {
+            boolean match = false;
+            for (BlackboardAttribute attribute : attributesList) {
+                BlackboardAttribute.Type attributeType = attribute.getAttributeType();
+                String attributeValue = attribute.getValueString();
```

Comment:
This seems to be ignoring the fact that not all attributes are Strings.

Reply:
I overlooked that point. Of course this also means the method signature will need to be changed to use "Object" instead of "String". I'll put together a change for this.